mypy/main.py

```diff
@@ -482,6 +527,48 @@ def add_invertible_flag(flag: str,
         print("Warning: --no-fast-parser no longer has any effect.  The fast parser "
               "is now mypy's default and only parser.")
 
+    # Infer Python version and/or executable if one is not given
+    if special_opts.python_executable is not None and special_opts.python_version is not None:
+        try:
+            py_exe_ver = _python_version_from_executable(special_opts.python_executable)
+
+        except PythonExecutableInferenceError as e:
+            parser.error(str(e))
+            sys.exit(2)
+        else:
+            if py_exe_ver != special_opts.python_version:
+                parser.error(
+                    'Error: Python version {} did not match executable {}, got version {}.'.format(
+                        special_opts.python_version, special_opts.python_executable, py_exe_ver
+                    ))
+                sys.exit(2)
+            else:
+                options.python_version = special_opts.python_version
+                options.python_executable = special_opts.python_executable
+    elif special_opts.python_executable is None and special_opts.python_version is not None:
+        options.python_version = special_opts.python_version
+        if not options.no_site_packages:
```

Comment:
I think `no_site_packages` should be in `special_opts`, because it's redundant information in `options` - indeed, you don't go on to check it outside this function